WcaOnRails/app/models/competition.rb

```diff
@@ -407,16 +430,8 @@ def latitude_radians
     end
   end
 
-  def events
-    # See https://github.com/cubing/worldcubeassociation.org/issues/95 for
-    # what these equal signs are about.
-    (eventSpecs || []).split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
-  end
-
   def has_events_with_ids?(event_ids)
-    # See https://github.com/cubing/worldcubeassociation.org/issues/95 for
-    # what these equal signs are about.
-    (event_ids - eventSpecs.split.map { |e| e.split("=")[0] }).empty?
+    (event_ids - events.ids).empty?
```

Comment:
Whatever you think is best.

On Tue, Sep 13, 2016, 14:43 Jeff Plumb <notifications@github.com> wrote:

> In WcaOnRails/app/models/competition.rb
> <https://github.com/cubing/worldcubeassociation.org/pull/759#discussion_r78650317>
> :
>
> >    def has_events_with_ids?(event_ids)
> > -    # See https://github.com/cubing/worldcubeassociation.org/issues/95 for
> > -    # what these equal signs are about.
> > -    (event_ids - eventSpecs.split.map { |e| e.split("=")[0] }).empty?
> > +    (event_ids - events.ids).empty?
>
> Hmmm, not in this in particular case since it's only used in the
> controller:
>
> @competitions = @competitions.select { |competition|
> competition.has_events_with_ids?(params[:event_ids]) }
>
> It's only if we called this procedure somewhere during the form cycle. So
> based on that should we keep it as it is, or update it in case it's ever
> used in such a scenario?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cubing/worldcubeassociation.org/pull/759/files/d27178e331cf7dd3c19d8f24bb8f6aa9318fccc9#r78650317>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAQ74o1olsLw9rksIW3F5ZtAVEaUO1d8ks5qpxj_gaJpZM4JDppm>
> .
>